pandas/core/indexes/base.py

```diff
@@ -3219,6 +3221,12 @@ def _convert_arr_indexer(self, keyarr):
         converted_keyarr : array-like
         """
         keyarr = com.asarray_tuplesafe(keyarr)
+
+        if needs_i8_conversion(self.dtype) and lib.infer_dtype(keyarr) == "string":
```

Comment:
instead of special-casing in the base class, let's override in `DatetimeIndexOpsMixin`